src/parsers/manifest/dash/common/indexes/timeline/timeline_representation_index.ts

```diff
@@ -402,25 +402,8 @@ export default class TimelineRepresentationIndex implements IRepresentationIndex
     if (this._index.timeline === null) {
       this._index.timeline = this._getTimeline();
     }
-
-    // destructuring to please TypeScript
-    const {
-      segmentUrlTemplate,
-      startNumber,
-      endNumber,
-      timeline,
-      timescale,
-      indexTimeOffset,
-    } = this._index;
     return getSegmentsFromTimeline(
-      {
-        segmentUrlTemplate,
-        startNumber,
-        endNumber,
-        timeline,
-        timescale,
-        indexTimeOffset,
-      },
+      this._index as typeof this._index & { timeline: IIndexSegment[] },
```

Comment:
Could timeline be initialized as `[]`  instead of `null` so it's easier ?

Reply:
I set its initial value as `null` on purpose, so that TypeScript can better catch when it is used without being unitialized.

Even if TypeScript is the one bothering us here